shaker/libs/pygit2_utils.py

```diff
@@ -0,0 +1,96 @@
+import shaker.libs.logger
+import pygit2
+import sys
+
+
+def pygit2_parse_error(e):
+    """
+    Parse a pygit2 specific error into a more understandable context. Will
+    also run some checks to try and help with the problem.
+
+    Args:
+        e(Exception): The exception that was raised
+    """
+    # Common errors to look for are,
+    # AttributeError: 'module' object has no attribute 'KeypairFromAgent'
+    # _pygit2.GitError: Unsupported URL protocol
+
+    # Set the logging level to noisy so we can generate as much usefult info as possible
+    shaker.libs.logger.Logger().basicConfig(level=shaker.libs.logger.Logger().INFO)
+    error_msg = ("shaker.libs.util:check_pygit2: Pygit2 problem detected, '%s'"
+                 % (e.message))
+    shaker.libs.logger.Logger().critical(error_msg)
+    pygit2_info()
+    shaker.libs.logger.Logger().critical("shaker.libs.util:check_pygit2: Running system checks for common issues")
+    pygit2_check()
+    sys.exit(1)
```

Comment:
Same thing here - sys.exit from a library is considered bad form.

We should throw a custom exception here – but only after checking `e` caused by what we expect. Right now this will stomp on any kind of exception with some logs then die which isn't very safe if the AttributeError we're called with isn't `'module' object has no attribute 'KeypairFromAgent'` – it could be hiding other errors.

Also we shouldn't be relying on the log output for providing diagnosis/helpful messages - make it part of the exception message we throw.